src/magnet_uri.cpp

```diff
@@ -215,8 +215,12 @@ namespace libtorrent {
 			url = uri.substr(pos, find(uri, "&", pos) - pos);
 		}
 
-		string_view btih = url_has_argument(uri, "xt");
-		if (btih.empty())
+		string_view btih = unescape_string(url_has_argument(uri, "xt"), ec);
```

Comment:
I would expect the vast majority of magnet links to not escape this. is there a simple way to only pay the cost of allocating a new string when you need to unescape it?